scripts/patch_release.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+# Copyright Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+MAJOR=$1
+MINOR=$2
+PATCH=$3
+
+NEW_VERSION="${MAJOR}.${MINOR}.${PATCH}"
+PREV_VERSION="${MAJOR}.${MINOR}.$((PATCH-1))"
+RELEASE_BRANCH="release-${MAJOR}.${MINOR}"
+
+git checkout "${RELEASE_BRANCH}"
+git pull "${ISTIOIO_GIT_SOURCE}" "${RELEASE_BRANCH}"
+
+echo "Migrating to the new release ${NEW_VERSION}..."
```

Comment:
Also there would be two different security related processes. One where the changes are in the public, so in that case we can update ref docs if we wanted (I suspect there wouldn't be any) and update the release version for the tests. In the case where the security build is private, neither of the changes would be required.

I would change SECURITY_PATCH to PRIVATE_PATCH and move the go mod and go tidy into the check with the update docs.